### ISSUE TITLE
Scanner does not pick up on friends list

### ISSUE THREAD

**Issue by ShyPixie** [PhantomGamers/SFP#4]
Since the old patcher has been deprecated, I'm forced to update to this patcher that never work since the first version. I already reset steam and everything.

log:

```[5/15/2022 5:47:44 PM][Info] Version is 0.0.4
[5/15/2022 5:47:44 PM][Info] Parsing cache...
[5/15/2022 5:47:44 PM][Info] Found 0 matches...
[5/15/2022 5:47:44 PM][Info] Scanner started
[5/15/2022 5:47:44 PM][Info] friends.css is already patched.
[5/15/2022 5:49:37 PM][Info] Resetting Steam
[5/15/2022 5:49:37 PM][Info] Shutting down Steam
[5/15/2022 5:49:39 PM][Info] Deleting c:\program files (x86)\steam\steamui\css
[5/15/2022 5:49:39 PM][Info] Deleting C:\Users\micro\AppData\Local\Steam\htmlcache\Cache
[5/15/2022 5:49:39 PM][Info] Starting Steam
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:17 PM][Info] Parsing cache...
[5/15/2022 5:50:17 PM][Info] Found 0 matches...
[5/15/2022 5:50:32 PM][Info] Parsing cache...
[5/15/2022 5:50:32 PM][Info] Found 0 matches...
```

this is the log when I use the old patcher (that works):
```[5/15/2022 5:57:32 PM] [Info] Checking for latest friends.css...
[5/15/2022 5:57:32 PM] [Info] Downloaded friends.css with etag: "D18GcwC82oVA"
[5/15/2022 5:57:33 PM] [Info] Downloaded friends.css with etag: "D18GcwC82oVA"
[5/15/2022 5:57:33 PM] [Info] Successfully downloaded friends.css files.
[5/15/2022 5:57:33 PM] [Info] Cache Watcher Started.
[5/15/2022 5:57:33 PM] [Info] Cache Watcher Stopped.
[5/15/2022 5:57:33 PM] [Info] Force scan started.
[5/15/2022 5:57:33 PM] [Info] Finding list of possible cache files...
[5/15/2022 5:57:33 PM] [Info] Found 73 possible cache files.
[5/15/2022 5:57:33 PM] [Info] Checking cache files for match...
[5/15/2022 5:57:33 PM] [Info] Successfully found matching friends.css for "D18GcwC82oVA" at C:\Users\micro\AppData\Local\Steam\htmlcache\Cache\f_000012.
[5/15/2022 5:57:33 PM] [Info] Overwriting with patched version...
[5/15/2022 5:57:33 PM] [Info] Reloading friends window...
[5/15/2022 5:57:34 PM] [Info] Done! Put your custom css in c:\program files (x86)\steam\clientui\friends.custom.css
[5/15/2022 5:57:34 PM] [Info] Close and reopen your Steam friends window to see changes.
[5/15/2022 5:57:34 PM] [Info] Cache Watcher Started.
```

current config:
```
{
  "userSettings": {
    "roaming": {
      "SFP.Properties.Settings": {
        "CacheDirectory": "C:\\Users\\micro\\AppData\\Local\\Steam\\htmlcache\\Cache",
        "CheckForUpdates": "False",
        "MinimizeToTray": "True",
        "RestartSteamOnPatch": "True",
        "ScanOnly": "False",
        "ShouldPatchFriends": "True",
        "ShouldPatchLibrary": "False",
        "ShouldPatchOnStart": "True",
        "ShouldScanFriends": "True",
        "ShouldScanLibrary": "False",
        "ShouldScanOnStart": "True",
        "ShowTrayIcon": "True",
        "StartMinimized": "True",
        "SteamDirectory": "c:\\program files (x86)\\steam",
        "SteamLaunchArgs": "-silent"
      }
    }
  }
}
```

**Comment by PhantomGamers** [comment]
Is that log from 0.0.4?

Is the behavior the same with 0.0.3? Or does it crash? Is the log the same in 0.0.3?

**Comment by ShyPixie** [comment]
I find out that the bug only occurs when SFP start itself before steam startup (due to #5), if I manually start steam and start SFP after that, everything is fine.

> Is that log from 0.0.4?
> 
yes

> Is the behavior the same with 0.0.3? Or does it crash? Is the log the same in 0.0.3?
> 
yes

> EDIT: Also can you try running the program as admin and see if that works? Which version of Windows are you running?
same as admin. Windows 11 (21H2 22000.675)

**Comment by PhantomGamers** [comment]
So just to clarify:

If you start SFP while Steam is not running and then start Steam, once Steam loads and the friends list opens neither the scanner nor the patch button work to patch the friends list until SFP is restarted while Steam is already running?

**Comment by ShyPixie** [comment]
> So just to clarify:
> 
> If you start SFP while Steam is not running and then start Steam, once Steam loads and the friends list opens neither the scanner nor the patch button work to patch the friends list until SFP is restarted while Steam is already running?

my friend list isn't set to automatically sign in when steam client starts. If a press patch button it does nothing. if a sign in on friends window it's not patched. if I press patch button after sign in on friends steam is restarted

**Comment by PhantomGamers** [comment]
This issue should be fixed in 0.0.6! :)

**Comment by PhantomGamers** [comment]
Closing this, feel free to reopen if you are still having this problem.
